web/client/actions/catalog.js

```diff
@@ -66,12 +66,14 @@ export const SET_FORMAT_OPTIONS = 'CATALOG:SET_FORMAT_OPTIONS';
  * Adds a list of layers from the given catalogs to the map
  * @param {string[]} layers list with workspace to be added in the map
  * @param {string[] | object[] } sources catalog names related to each layer
+ * @param {string[]} filters filters related to each layer
  */
-export function addLayersMapViewerUrl(layers = [], sources = []) {
+export function addLayersMapViewerUrl(layers = [], sources = [], filters = []) {
```

Comment:
can you change filters to be params?

Reply:
I said `options` so we can override whatever we want. The final layer have to be simply a deep merge of the 2 JSON

e.g. `options=[{"params":{"CQL_FILTER":"A=B"}}, ...]`

Reply:
Thanks! Updated code to use "options" argument.